WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -491,175 +556,104 @@ class PostListFragment : Fragment(),
-        if (event.causeOfChange is CauseOfOnPostChanged.UpdatePost) {
-            // if a Post is updated, let's refresh the whole list, because we can't really know
-            // from FluxC which post has changed, or when. So to make sure, we go to the source,
-            // which is the FluxC PostStore.
-            /*
-             * Please note that the above comment is no longer correct as `CauseOfOnPostChanged.UpdatePost` has fields
-             * for local and remote post id. However, this class will be completely refactored in an upcoming PR, so
-             * both the original comment and this note is kept for record keeping.
-             */
-            if (!event.isError) {
-                loadPosts(LoadMode.IF_CHANGED)
+    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @Suppress("unused")
+    fun onListChanged(event: OnListChanged) {
+        if (!event.listDescriptors.contains(listDescriptor)) {
+            return
+        }
+        if (event.isError) {
+            GlobalScope.launch(Dispatchers.Main) {
+                val emptyViewMessageType = if (event.error.type == ListErrorType.PERMISSION_ERROR) {
+                    PERMISSION_ERROR
+                } else GENERIC_ERROR
+                updateEmptyView(emptyViewMessageType)
             }
-        } else if (event.causeOfChange is CauseOfOnPostChanged.FetchPosts) {
-            isFetchingPosts = false
-            if (!isAdded) {
-                return
+        } else {
+            if (event.causeOfChange == FIRST_PAGE_FETCHED) {
+                // `uploadedPostRemoteIds` is kept as a workaround when the local drafts are uploaded and the list
+                // has not yet been updated yet. Since we just fetched the first page, we can safely clear it.
+                // Please check out `onPostUploaded` for more context.
+                uploadedPostRemoteIds.clear()
             }
+            refreshListManagerFromStore(listDescriptor, false)
+        }
+    }
 
-            hideLoadMoreProgress()
-            if (!event.isError) {
-                canLoadMorePosts = event.canLoadMore
-                loadPosts(LoadMode.IF_CHANGED)
-            } else {
-                val error = event.error
-                when (error.type) {
-                    PostStore.PostErrorType.UNAUTHORIZED -> updateEmptyView(EmptyViewMessageType.PERMISSION_ERROR)
-                    else -> updateEmptyView(EmptyViewMessageType.GENERIC_ERROR)
-                }
-            }
-        } else if (event.causeOfChange is CauseOfOnPostChanged.DeletePost) {
-            if (event.isError) {
-                val message = getString(R.string.error_deleting_post)
-                ToastUtils.showToast(nonNullActivity, message, ToastUtils.Duration.SHORT)
-                loadPosts(LoadMode.IF_CHANGED)
-            }
+    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @Suppress("unused")
+    fun onListItemsChanged(event: OnListItemsChanged) {
```

Comment:
Whenever there is a change for a specific "type", FluxC will emit this event. In this case, the type will represent the posts belonging to a specific site. We need to check if the type that's updated is the one we are interested in and update the views accordingly.

The only type we have in FluxC is for posts right now. However, once we have more types, this will be an important check. Also, if you switch the site, the "type" will actually change, so if a post for another site is updated, the list will not be refreshed, as expected.